### DISCUSSION TITLE
How to create a floating point term from a floating point

### DISCUSSION THREAD

**Discussion by thealberto** [cvc5/cvc5#10748] [Q&A] [answered]
Hi All,
I'm trying to understand how to create a floating point `Term` from a floating point in C++. Currently I can do it using the following pseudo code:

```
Term *getFloat(float n) {
  bitset<sizeof(float) * 8> b(numberToBitString(n)); 
  Term v =  slv->mkBitVector(32, b.to_ullong());
  Term fpv = slv->mkFloatingPoint(8,24,v);
  return fpv;
}
```

I feel it is a bit convoluted and there must be a better way.

Any suggestion?

Thanks

**Comment by mpreiner** [comment, marked as answer]
We currently don't provide functions for creating FP values from float/double. However, this is definitely a feature we can add to the API.  I created an issue to track this feature: https://github.com/cvc5/cvc5/issues/10762
